tools/wptrunner/wptrunner/browsers/webkit.py

```diff
@@ -0,0 +1,91 @@
+from .base import Browser, ExecutorBrowser, require_arg
+from ..executors import executor_kwargs as base_executor_kwargs
+from ..executors.executorselenium import (SeleniumTestharnessExecutor,
+                                          SeleniumRefTestExecutor)
+from ..executors.executorwebkit import WebKitDriverWdspecExecutor
+from ..webdriver_server import WebKitDriverServer
+
+
+__wptrunner__ = {"product": "webkit",
+                 "check_args": "check_args",
+                 "browser": "WebKit",
+                 "browser_kwargs": "browser_kwargs",
+                 "executor": {"testharness": "SeleniumTestharnessExecutor",
+                              "reftest": "SeleniumRefTestExecutor",
+                              "wdspec": "WebKitDriverWdspecExecutor"},
+                 "executor_kwargs": "executor_kwargs",
+                 "env_extras": "env_extras",
+                 "env_options": "env_options"}
+
+
+def check_args(**kwargs):
+    require_arg(kwargs, "binary")
+    require_arg(kwargs, "webdriver_binary")
+    require_arg(kwargs, "webkit_port")
+
+
+def browser_kwargs(test_type, run_info_data, **kwargs):
+    return {"binary": kwargs["binary"],
+            "webdriver_binary": kwargs["webdriver_binary"],
+            "webdriver_args": kwargs.get("webdriver_args")}
+
+
+def capabilities_for_port(webkit_port, binary, binary_args):
+    if webkit_port == "gtk":
+        return {"browserName": "MiniBrowser",
```

Comment:
What is MiniBrowser?

Reply:
It's a small prototyping browser application that's used for quick tests on a WebKit build. As such, it's also usable as a WebDriver automation target.

Reply:
I found https://github.com/WebKit/webkit/tree/master/Tools/MiniBrowser. So it's not specific to the GTK port, but multiple ports have this "product"?

Reply:
Correct. Implementations are different, but in principle it's a limited Web view application, not a full-fledged browsing experience.

There's actually a DesiredCapabilities.WEBKITGTK dict already added to the Selenium project, it could be used here directly like it is for other products, instead of being replicated.

Reply:
I'm not clear on how the capabilities are used, but consistency sounds good.